File: src/types/volunteer.ts
export interface Volunteer {
  id: string;
  name: string;
  email: string;
  phone: string;
  location: string;
  interest: string;
  message?: string;
  created_at: string;
}

export interface VolunteerFormData {
  full_name: string;
  email: string;
  phone: string;
  date_of_birth: string;
  address: string;
  city: string;
  country: string;
  education_level: string;
  occupation: string;
  skills: string;
  availability: string;
  previous_experience: string;
  motivation: string;
  emergency_contact_name: string;
  emergency_contact_phone: string;
}

File: src/main/java/org/hearthy/model/Volunteer.java
package org.hearthy.model;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.OffsetDateTime;
import java.util.UUID;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class Volunteer {
    private UUID id;
    private String name;
    private String email;
    private String phone;
    private String location;
    private String interest;
    private String message;
    private OffsetDateTime createdAt;
}
